cplusplus/include/vips/VImage8.h

```diff
@@ -222,10 +222,10 @@ class VOption
 	VOption *set( const char *name, int value );
 	VOption *set( const char *name, double value );
 	VOption *set( const char *name, const char *value );
-	VOption *set( const char *name, const VImage &value );
-	VOption *set( const char *name, const VInterpolate &value ); 
-	VOption *set( const char *name, const VStreamI &value );
-	VOption *set( const char *name, const VStreamO &value );
+	VOption *set( const char *name, const VImage value );
```

Comment:
`const` could be removed, since it doesn't affect the signature of a function. See:
https://clang.llvm.org/extra/clang-tidy/checks/readability-avoid-const-params-in-decls.html

(if you want, I can make a PR to fix all `clang-tidy` linter errors)

Reply:
VImage etc. are really smart pointers, so the const is suppsed to suggest to users that the implicit reference is const. Perhaps it's not helpful :(

Reply:
Top-level `const` is very useful within the implementation (i.e., code in the body of the function) and the compiler will enforce it, however it is ignored from the interface viewpoint. For example:
```cpp
VOption *set( const char *name, const VImage value );
```
and
```cpp
VOption *set( const char *name, VImage value );
```
declares the same function.

(https://stackoverflow.com/q/17208570/10952119 might be relevant here)

Reply:
Ye exactly, call-by-value is always const. My idea was to add the const decorators for human readers to try to make it plain that the underlying image was not being modified.

Reply:
Ah I see, that makes sense. As long as there are no compiler warnings, I'm fine with it.

fwiw, I've looked at the rest of the warnings and think they're safe to ignore. I only noticed that we could replace a `memset` and initialize a GValue directly within the member initialization list:
```diff
--- cplusplus/include/vips/VImage8.h
+++ cplusplus/include/vips/VImage8.h
@@ -197,10 +197,9 @@
 		}; 
 
 		Pair( const char *name ) : 
-			name( name ), input( false ), vimage( 0 )
+			name( name ), input( false ), vimage( 0 ), 
+			value( G_VALUE_INIT )
 		{
-			// argh = {0} won't work wil vanilla C++
-			memset( &value, 0, sizeof( GValue ) ); 
 		}
 
 		~Pair()
```

Unfortunately the `G_VALUE_INIT` macro is only available since GLib 2.30 :(